lib/Sema/Constraint.cpp

```diff
@@ -324,8 +327,15 @@ void Constraint::dump(ConstraintSystem *CS) const {
   // Print all type variables as $T0 instead of _ here.
   llvm::SaveAndRestore<bool> X(CS->getASTContext().LangOpts.
                                DebugConstraintSolver, true);
-  
+  // Disabled MSVC warning: only for use within the debugger
+#if defined(_MSC_VER)
+#pragma warning(push)
+#pragma warning(disable: 4996)
+#endif
   dump(&CS->getASTContext().SourceMgr);
+#if defined(_MSC_VER)
+#pragma warning(pop)
+#endif
 }
```

Comment:
I'm really curious why this is only hitting here, and only with VC++. Do you know if clang transitively decides not to warn here because it's being called from another deprecated function?